core-bundle/src/Resources/contao/library/Contao/Controller.php

```diff
@@ -77,9 +77,10 @@ abstract class Controller extends System
 	public static function getTemplate($strTemplate)
 	{
 		$strTemplate = basename($strTemplate);
+		$request = System::getContainer()->get('request_stack')->getCurrentRequest();
 
 		// Check for a theme folder
-		if (\defined('TL_MODE') && TL_MODE == 'FE')
+		if ($request && System::getContainer()->get('contao.routing.scope_matcher')->isFrontendRequest($request))
```

Comment:
This change is unrelated, isn‘t it?

Reply:
No, without it the change will not work as `TL_MODE` will be `'BE'`.

Reply:
This PR fixes #5552. This issue occurs if you are using a custom template in a theme subfolder, e.g. `foobar/ce_text_foobar.html5` for example and reference that template in one of your content elements of one of your news for which you have enabled the news feed. The news feed XML is then generated by Contao within the back end scope, which causes various issues (including #5552). For this reason we introduced a faux request stack in order to simulate a front end request and also simulate a `$GLOBALS['objPage']` object for the respective request.

This PR fixes #5552 by also setting the `templateGroup` on the global page object. However, this change alone will not fix it, as `Controller::getTemplate` (which will be called during rendering of a content element) ignored the `templateGroup` if `TL_MODE` is not `FE`. Thus this PR also needs to change this logic to using the ScopeMatcher instead.

Reply:
Ah, now I see. 👍

Reply:
> `Controller::getTemplate` (which will be called during rendering of a content element) ignores the `templateGroup` if `TL_MODE` is not `'FE'`.

That seems correct to me, as template groups are only supported in the front end.

Reply:
Yes - hence the change in this Pull Request. Or what is your point?